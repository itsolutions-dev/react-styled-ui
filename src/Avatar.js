// @flow

import React from 'react';
import styled, { css } from './styledComponent';
import {
  getSizeOrDefault,
} from './utils/';

const defaultSize = '40px';
const defaultRadius = '99999px';

type AvatarProps = {
  src: string,
  alt: string,
  children: any,
};

const Avatar = styled({
  component: 'div',
  props: {
    children: ({ src, alt, children }: AvatarProps) => (
      src == null ? children : <img src={src} alt={alt || src} />
    ),
  },
  style: css`    
    ${props => (props.src == null ? `
      width: ${getSizeOrDefault(props.size, defaultSize)};
      height: ${getSizeOrDefault(props.size, defaultSize)};
      border-radius: ${getSizeOrDefault(props.radius, defaultRadius)}; 
      color: ${(props.color == null ? 'rgb(255, 255, 255)' : props.color)};
      background-color: ${(props.backgroundColor == null ? 'rgb(188, 188, 188)' : props.backgroundColor)} !important;
      font-size: ${getSizeOrDefault(props.fontSize, '20px')};
      user-select: none;
      align-items: center;
      justify-content: center;
      display: inline-flex;
    ` : `
      > img {
        width: ${getSizeOrDefault(props.size, defaultSize)};
        height: ${getSizeOrDefault(props.size, defaultSize)};
        border-radius: ${getSizeOrDefault(props.radius, defaultRadius)};
        display: inline-flex;
      } 
    `)} 
  `,
});

export default Avatar;
