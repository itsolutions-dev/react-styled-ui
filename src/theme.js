export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
