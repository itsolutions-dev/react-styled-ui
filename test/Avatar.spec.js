import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Avatar from '../src/Avatar';

describe('Avatar', () => {
  test('should render a Img-Avatar', () => {
    const component = ReactTestRenderer.create(<Avatar src="http://lorempixel.com/400/200/nature/" theme={theme} />);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have an img tag [IMG]', () => {
    const component = ReactTestRenderer.create(<Avatar src="http://lorempixel.com/400/200/nature/" theme={theme} />).toJSON();
    expect(component.children[0].type).toEqual('img');
  });

  test('should have a default radius [IMG]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        src="http://lorempixel.com/400/200/nature/"
        theme={theme}
      />,
    );
    expect({
      component,
      modifier: '> img',
    }).toHaveStyleRule('border-radius', '99999px');
  });

  test('should have a radius [IMG]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        src="http://lorempixel.com/400/200/nature/"
        theme={theme}
        radius={40}
      />,
    );
    expect({
      component,
      modifier: '> img',
    }).toHaveStyleRule('border-radius', '40px');
  });

  test('should have a default size [IMG]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        src="http://lorempixel.com/400/200/nature/"
        theme={theme}
      />,
    );
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('height', '40px');
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('width', '40px');
  });

  test('should handle string size [IMG]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        src="http://lorempixel.com/400/200/nature/"
        theme={theme}
        size="20em"
      />,
    );
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('height', '20em');
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('width', '20em');
  });

  test('should handle numeric size [IMG]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        src="http://lorempixel.com/400/200/nature/"
        theme={theme}
        size={30}
      />,
    );
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('height', '30px');
    expect(({
      component,
      modifier: '> img',
    })).toHaveStyleRule('width', '30px');
  });

  test('should render a Avatar [TXT]', () => {
    const component = ReactTestRenderer.create(<Avatar theme={theme} >A</Avatar>);
    expect(component).toBeDefined();
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should have a div tag [TXT]', () => {
    const component = ReactTestRenderer.create(<Avatar theme={theme} >A</Avatar>).toJSON();
    expect(component.type).toEqual('div');
  });

  test('should have a default radius [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar theme={theme}>A</Avatar>,
    );
    expect(component).toHaveStyleRule('border-radius', '99999px');
  });

  test('should have a radius [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        radius={40}
      >
        A
      </Avatar>,
    );
    expect(component).toHaveStyleRule('border-radius', '40px');
  });

  test('should have a default size [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar theme={theme}>A</Avatar>,
    );
    expect(component).toHaveStyleRule('height', '40px');
    expect(component).toHaveStyleRule('width', '40px');
  });

  test('should handle string size [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        size="20em"
      >
        A
      </Avatar>,
    );
    expect(component).toHaveStyleRule('height', '20em');
    expect(component).toHaveStyleRule('width', '20em');
  });

  test('should handle numeric size [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        size={30}
      >
        A
      </Avatar>,
    );
    expect(component).toHaveStyleRule('height', '30px');
    expect(component).toHaveStyleRule('width', '30px');
  });

  test('should have color [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        color="#FF0000"
      >
        A
      </Avatar>,
    );
    expect(component).toHaveStyleRule('color', '#FF0000');
  });

  test('should have background-color [TXT]', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        backgroundColor="#FF0000"
      >
        A
      </Avatar>,
    );
    expect(component).toHaveStyleRule('background-color', '#FF0000 !important');
  });

  test('should match snapshot', () => {
    const component = ReactTestRenderer.create(
      <Avatar
        theme={theme}
        position="absolute"
      >
        A
      </Avatar>,
    );
    expect(component).toMatchSnapshot();
  });
});
