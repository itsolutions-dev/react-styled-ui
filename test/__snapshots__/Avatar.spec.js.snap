// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render a Avatar 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  border-radius: 99999px;
}

<img
  className="c0"
/>
`;
